### PR TITLE
Implemented graphql service

### PR DESCRIPTION
**Pull request by jluhrs**
Basic implementation has two TCS commands in the form of mutations.

### CODE REVIEW

File: modules/web/server/src/main/scala/engage/web/server/http4s/EngageMappings.scala [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+class EngageMappings[F[_]: Sync](server: EngageEngine[F])(override val schema: Schema)
+    extends ValueMapping[F] {
+  import EngageMappings._
+
+  def mountPark(p: Path, env: Cursor.Env): F[Result[Cursor]] =
+    server.mcsPark.attempt.map(
+      _.bimap(
+        e => NonEmptyChain(Problem(e.getMessage)),
+        _ => valueCursor(p, env, ParkStatus.Parked)
```
Comment (jluhrs) [inline modules/web/server/src/main/scala/engage/web/server/http4s/EngageMappings.scala:42]: This is temporary, to be replaced by real value read from TCS.

File: modules/web/server/src/main/scala/engage/web/server/http4s/EngageMappings.scala [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  def mountFollow(p: Path, env: Cursor.Env): F[Result[Cursor]] =
+    env
+      .get[Boolean]("enable")
+      .map(server.mcsFollow)
+      .getOrElse(Applicative[F].unit)
+      .attempt
+      .map(
+        _.bimap(
+          e => NonEmptyChain(Problem(e.getMessage)),
+          _ => valueCursor(p, env, FollowStatus.Following)
```
Comment (jluhrs) [inline modules/web/server/src/main/scala/engage/web/server/http4s/EngageMappings.scala:55]: This is temporary, to be replaced by real value read from TCS.

File: modules/web/server/src/main/scala/engage/web/server/http4s/GraphQlRoutes.scala [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import org.http4s.server.middleware.GZip
+import org.http4s.server.websocket.WebSocketBuilder2
+import org.typelevel.log4cats.Logger
+
+class GraphQlRoutes[F[_]: Async: Logger: Trace](
+  eng: EngageEngine[F]
+) extends Http4sDsl[F] {
+
+  private def commandServices(wsb: WebSocketBuilder2[F]): HttpRoutes[F] = GZip(
+    Routes.forService(_ => EngageMappings(eng).map(GrackleGraphQLService[F](_).some), wsb, "engage")
```
Comment (jluhrs) [inline modules/web/server/src/main/scala/engage/web/server/http4s/GraphQlRoutes.scala:28]: I couldn't figure out how to use `AuthRoutes` here.

File: build.sbt [hunk 4]
```diff
@@ -133,95 +126,6 @@ lazy val engage_web_server = project
   .dependsOn(engage_server)
   .dependsOn(engage_model.jvm % "compile->compile;test->test")
 
-lazy val engage_web_client = project
```
Comment (cquiroz) [inline build.sbt:136]: Good, this will be done separetely

**Review by cquiroz — APPROVED** [review]
LGTM, thanks a lot

File: modules/web/server/src/main/scala/engage/web/server/http4s/GraphQlRoutes.scala (hunk 3 above)
Comment (tpolecat) [reply modules/web/server/src/main/scala/engage/web/server/http4s/GraphQlRoutes.scala:28]: No `AuthedRoutes` here,  use lucuma-sso-client to do the authentication. Here is how the odb does it: https://github.com/gemini-hlsw/lucuma-odb/blob/main/modules/service/src/main/scala/lucuma/odb/graphql/GraphQLRoutes.scala

File: modules/web/server/src/main/scala/engage/web/server/http4s/EngageMappings.scala [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+  val MutationType: TypeRef     = schema.ref("Mutation")
+  val ParkStatusType: TypeRef   = schema.ref("ParkStatus")
+  val FollowStatusType: TypeRef = schema.ref("FollowStatus")
+
+  override val typeMappings: List[TypeMapping] = List(
+    ObjectMapping(
+      tpe = MutationType,
+      fieldMappings = List(
+        RootEffect.computeCursor("mountPark")((_, p, env) => mountPark(p, env)),
+        RootEffect.computeCursor("mountFollow")((_, p, env) => mountFollow(p, env))
```
Comment (tpolecat) [inline modules/web/server/src/main/scala/engage/web/server/http4s/EngageMappings.scala:68]: This looks fine :+1:

File: modules/web/server/src/main/scala/engage/web/server/http4s/GraphQlRoutes.scala (hunk 3 above)
Comment (cquiroz) [reply modules/web/server/src/main/scala/engage/web/server/http4s/GraphQlRoutes.scala:28]: It is not a given that engage will use Lucuma-sso, it is for internal users authenticating with noirlab sso
I guess it is open if we'll allow login via orcid

File: modules/web/server/src/main/resources/NewTCC.graphql [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+type MountStatus {
+    parked: ParkStatus!
+    follow: FollowStatus!
+}
+
+type TelescopeStatus {
+    mount: MountStatus!
+}
+
+type Query {
```
[the remark this answers is not in the document]
Comment (jluhrs) [reply modules/web/server/src/main/resources/NewTCC.graphql:20]: Yes, that is the idea.

File: modules/web/server/src/main/scala/engage/web/server/http4s/WebServerLauncher.scala [hunk 7]
```diff
@@ -125,7 +126,8 @@ object WebServerLauncher extends IOApp with LogInitialization {
       "/"                    -> new StaticRoutes(conf.mode === Mode.Development, OcsBuildInfo.builtAtMillis).service,
       "/api/engage/commands" -> new EngageCommandRoutes(as, se).service,
```
[the remark this answers is not in the document]
Comment (jluhrs) [reply modules/web/server/src/main/scala/engage/web/server/http4s/WebServerLauncher.scala:127]: They will become obsolete, eventually.

**Pull request merged**